plugins/services/src/js/data/MarathonClient.ts

```diff
@@ -0,0 +1,53 @@
+import { request } from "@dcos/http-service";
+
+import Config from "#SRC/js/config/Config";
+import { map } from "rxjs/operators";
+import { Observable } from "rxjs";
+
+function buildMarathonURI(path: string) {
+  return `${Config.rootUrl}${Config.marathonAPIPrefix}${path}`;
+}
+
+interface CreateGroupErrorResponse {
+  message: string;
+  details?: Array<{ path: string; errors: string[] }>;
+}
+
+export interface CreateGroupResponse {
+  deploymentId: string;
+  version: string;
+}
+
+export function createGroup(
+  id: string,
+  enforceRole: boolean
+): Observable<CreateGroupResponse> {
+  return request(buildMarathonURI("/groups"), {
+    method: "POST",
+    body: JSON.stringify({ id, enforceRole })
+  }).pipe(
+    map(reqResp => {
```

Comment:
This appears to be called for success, but not for some errors (duplicate group name).

Reply:
yes that is expected, an error will exit the stream and if there is an error creating the group we don't want to create quota.